org-roam-db.el

```diff
@@ -254,6 +254,20 @@ This is equivalent to removing the node from the graph."
                            file
                            :limit 1)))
 
+(defun org-roam-db--connected-component (file)
+  "Return all files reachable from/connected to FILE."
+  (let* ((query "WITH RECURSIVE
+                   links_of(file, link) AS
+                     (SELECT file_from, file_to FROM file_links UNION
+                      SELECT file_to, file_from FROM file_links),
+                   connected_component(file) AS
+                     (SELECT link FROM links_of WHERE file = $s1
+                      UNION
+                      SELECT link FROM links_of JOIN connected_component USING(file))
+                   SELECT * FROM connected_component;")
```

Comment:
Unfortunately I was not able to express two CTEs in a single query using the emacsql s-expressions.

Reply:
this is fine.